web/components/notifications/notification-types.tsx

```diff
@@ -0,0 +1,899 @@
+      />
+    )
+  } else if (sourceType === 'user') {
+    return (
+      // not appearing?
+      <UserJoinedNotification
+        notification={notification}
+        isChildOfGroup={isChildOfGroup}
+        highlighted={highlighted}
+      />
+    )
+  } else if (sourceType === 'challenge') {
+    return (
+      <ChallengeNotification
+        notification={notification}
+        isChildOfGroup={isChildOfGroup}
+        highlighted={highlighted}
+      />
+    )
+  }
+  return (
+    <NotificationFrame
+      notification={notification}
+      highlighted={highlighted}
+      isChildOfGroup={isChildOfGroup}
+      icon={<></>}
+    >
+      <div className={'mt-1 ml-1 md:text-base'}>
+        <NotificationTextLabel notification={notification} />
+      </div>
+    </NotificationFrame>
+  )
+}
+
+function BetFillNotification(props: {
+  notification: Notification
+  highlighted: boolean
+  isChildOfGroup?: boolean
+}) {
+  const { notification, isChildOfGroup, highlighted } = props
+  const { sourceText, data, sourceContractTitle } = notification
+  const { creatorOutcome, probability, limitOrderRemaining } =
+    (data as BetFillData) ?? {}
+  const amount = formatMoney(parseInt(sourceText ?? '0'))
+  const color =
+    creatorOutcome === 'YES'
+      ? 'text-teal-600'
+      : creatorOutcome === 'NO'
+      ? 'text-scarlet-600'
+      : 'text-blue-600'
+  const description =
+    creatorOutcome && probability ? (
+      <span>
+        {amount} of your
+        <span className={clsx('mx-1', color)}>{creatorOutcome}</span>
+        limit order at was filled{' '}
+      </span>
+    ) : (
+      <span>{amount} of your limit order was filled</span>
+    )
+
+  const subtitle = (
+    <>
+      {limitOrderRemaining ? (
+        <>
+          Your limit order will buy{' '}
+          <span className={clsx(color)}>{creatorOutcome}</span> down to{' '}
+          <b>{Math.round(probability * 100)}%</b>. You have{' '}
+          {formatMoney(limitOrderRemaining)} remaining.
+        </>
+      ) : (
+        ''
+      )}
+    </>
+  )
+
+  return (
+    <NotificationFrame
+      notification={notification}
+      isChildOfGroup={isChildOfGroup}
+      highlighted={highlighted}
+      icon={
+        <NotificationIcon
+          symbol={creatorOutcome === 'NO' ? '👇' : '☝️'}
+          symbolBackgroundClass={
+            creatorOutcome === 'NO'
+              ? 'bg-gradient-to-br from-scarlet-600 to-scarlet-300'
+              : 'bg-gradient-to-br from-teal-600 to-teal-300'
+          }
+        />
+      }
+      subtitle={subtitle}
+      link={getSourceUrl(notification)}
+    >
+      <>
+        {description}
+        {!isChildOfGroup && (
+          <span>
+            on{' '}
+            <PrimaryNotificationLink
+              text={sourceContractTitle}
+              truncatedLength={'md'}
+            />
+          </span>
+        )}
+      </>
+    </NotificationFrame>
+  )
+}
+
+function BadgeNotification(props: {
```

Comment:
I no longer award badges as a matter of fact so it'd be good to remove this component

Reply:
shouldn't we keep it for older badge notifications though?

Reply:
Oh yeah I suppose we can keep it in for another month and then remove it, good thinking.